src/webgl/shaders/lighting.glsl

```diff
@@ -67,6 +79,59 @@ LightResult _light(vec3 viewDirection, vec3 normal, vec3 lightVector) {
   return lr;
 }
 
+// converts the range of "value" from [min1 to max1] to [min2 to max2]
+float map(float value, float min1, float max1, float min2, float max2) {
+  return min2 + (value - min1) * (max2 - min2) / (max1 - min1);
+}
+
+vec2 mapTextureToNormal( vec3 normal ){
+  // x = r sin(phi) cos(theta)
+  // y = r cos(phi)
+  // z = r sin(phi) sin(theta)
+  float phi = acos( normal.y );
+  // if phi is 0, then there are no x, z components
+  float theta = 0.0;
+  // else 
+  if(  abs(sin(phi)) > 0.0 ){
+    theta = acos( normal.x / sin(phi) );
+  }   
+
+  vec2 newTexCoor = vec2(
+    map(theta, 0., PI, 0., 1.),
+    map(phi, 0., PI, 1., 0.)
+  );
+
+  return newTexCoor;
+}
+
+vec3 calculateImageDiffuse( vec3 vNormal, vec3 vViewPosition ){
+  // make 2 seperate builds 
+  vec3 worldCameraPosition =  vec3(0.0, 0.0, 0.0);  // hardcoded world camera position
+  vec3 worldNormal = normalize(vNormal);
+  vec3 lightDirection = normalize( vViewPosition - worldCameraPosition );
+  vec3 R = reflect(lightDirection, worldNormal);
+  // use worldNormal instead of R
+  vec2 newTexCoor = mapTextureToNormal( R );
+  vec4 texture = TEXTURE( environmentMapDiffused, newTexCoor );
+  return texture.xyz;
+}
+
+vec3 calculateImageSpecular( vec3 vNormal, vec3 vViewPosition ){
+  vec3 worldCameraPosition =  vec3(0.0, 0.0, 0.0);
+  vec3 worldNormal = normalize(vNormal);
+  vec3 lightDirection = normalize( vViewPosition - worldCameraPosition );
+  vec3 R = reflect(lightDirection, worldNormal);
+  vec2 newTexCoor = mapTextureToNormal( R );
+#ifdef WEBGL2
+  vec4 outColor = textureLod(environmentMapSpecular, newTexCoor, levelOfDetail);
+#else
+  vec4 outColor = TEXTURE(environmentMapSpecular, newTexCoor);
+#endif
+  // this is to make the darker sections more dark
+  // png and jpg usually flatten the brightness so it is to reverse that
+  return pow(outColor.xyz, 10.0);
```

Comment:
Looks like tests are failing because `pow` needs both parameters to be the same type :') So maybe this needs to be:
```suggestion
  return pow(outColor.xyz, vec3(10.0));
```